### PR TITLE
Task feature implemented with some bugs to resolve.

### PR DESCRIPTION
**Pull request by SunnyKhandare**
I have implemented task feature but there are some bugs to resolve.

### CODE REVIEW

File: src/main/java/com/todoapp/app/entity/Task.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    @NotBlank( message = "Task name is necessary" )
+    private String task_name;
+
+    private long task_creation_date;
+
+    private long deadline;
+
+    private String task_description;
+
```
Comment (dakshpareek) [inline src/main/java/com/todoapp/app/entity/Task.java:25]: Change task_creation_date and deadline type to Date

File: src/main/java/com/todoapp/app/entity/Task.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    private String task_status;
+
+    public Task(){
+
+    }
+
+    public Task(User user, @NotBlank(message = "Task name is necessary") String task_name) {
```
Comment (dakshpareek) [inline src/main/java/com/todoapp/app/entity/Task.java:32]: No need for multiple constructors, use just one more other than default one.

File: src/main/java/com/todoapp/app/restcontroller/TaskResource.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    @PostMapping( "/users/{user_id}/tasks" )
+    public void add_task(@PathVariable int id , @RequestBody Task task ){
+
+        task_repository.save( task );
+    }
+
+    @DeleteMapping( "/users/{user_id}/task/{task_id}" )
+    public void delete_pending_task( @PathVariable int user_id , @PathVariable int task_id ){
+
```
Comment (dakshpareek) [inline src/main/java/com/todoapp/app/restcontroller/TaskResource.java:31]: How you are making sure that this task is related to this provided user not other user ? What if I try to delete a task by giving a task id but it belongs to other users ?

**Review by dakshpareek — CHANGES_REQUESTED** [review]
Please check out the comments and make changes accordingly.

**Comment by dakshpareek** [comment]
make changes

File: src/main/java/com/todoapp/app/restcontroller/TaskResource.java (hunk 3 above)
Comment (SunnyKhandare) [reply src/main/java/com/todoapp/app/restcontroller/TaskResource.java:31]: how an user will know the task id of tasks of other users.  And task_id is an primary key which will be unique in task table.

Comment (SunnyKhandare) [reply src/main/java/com/todoapp/app/restcontroller/TaskResource.java:31]: maaf karde yaar. Galtii hoo gayiiii

**Pull request merged**